src/test/java/ch/wisv/areafiftylan/TokenAuthenticationTest.java

```diff
@@ -0,0 +1,225 @@
+    AuthenticationTokenRepository authenticationTokenRepository;
+
+    @After
+    public void cleanupAuthTest() {
+        authenticationTokenRepository.deleteAll();
+    }
+
+    private Header getTokenHeader(String username, String password) {
+        Map<String, String> userDTO = new HashMap<>();
+        userDTO.put("username", username);
+        userDTO.put("password", password);
+
+        //@formatter:off
+        String token =
+            given().
+            when().
+                content(userDTO).contentType(ContentType.JSON).
+                post("/token").
+            then().
+                extract().path("object");
+        //@formatter:on
+
+        return new Header("X-Auth-Token", token);
+    }
+
+    @Test
+    public void testRequestToken() {
+        Map<String, String> userDTO = new HashMap<>();
+        userDTO.put("username", user.getUsername());
+        userDTO.put("password", userCleartextPassword);
+
+        //@formatter:off
+        given().
+        when().
+            content(userDTO).contentType(ContentType.JSON).
+            post("/token").
+        then().
+            statusCode(HttpStatus.SC_OK);
+        //@formatter:on
+
+        AuthenticationToken authenticationToken1 =
+                authenticationTokenRepository.findByUserUsername(user.getUsername()).orElse(null);
+
+        //@formatter:off
+        given().
+        when().
+            content(userDTO).contentType(ContentType.JSON).
+            post("/token").
+        then().
+            statusCode(HttpStatus.SC_OK);
+        //@formatter:on
+
+        AuthenticationToken authenticationToken2 =
+                authenticationTokenRepository.findByUserUsername(user.getUsername()).orElse(null);
+
+        Assert.assertFalse(authenticationToken1.equals(authenticationToken2));
+    }
+
+    @Test
+    public void testRequestTokenRenew() {
+        Map<String, String> userDTO = new HashMap<>();
+        userDTO.put("username", user.getUsername());
+        userDTO.put("password", userCleartextPassword);
+
+        //@formatter:off
+        given().
+        when().
+            content(userDTO).contentType(ContentType.JSON).
+            post("/token").
+        then().
+            statusCode(HttpStatus.SC_OK);
+        //@formatter:on
+
+        Optional<AuthenticationToken> authenticationToken =
+                authenticationTokenRepository.findByUserUsername(user.getUsername());
+
+        Assert.assertTrue(authenticationToken.isPresent());
+
+    }
+
+    @Test
+    public void testVerifyValidToken() {
+        Header header = getTokenHeader(user.getUsername(), userCleartextPassword);
+
+        //@formatter:off
+        given().
+            header(header).
+        when().
+            post("/token/verify").
+        then().
+            statusCode(HttpStatus.SC_OK);
+        //@formatter:on
+    }
+
+    @Test
+    public void testVerifyInvalidToken() {
+        Header header = new Header("X-Auth-Token", "invalid");
```

Comment:
I personally would have stored this header key in a private final in this class. These are magic strings. Same goes for the invalid header.